### ISSUE TITLE
Question: Access underlying window by handle

### ISSUE THREAD

**Issue by mar1ged** [fyne-io/fyne#476]
I know that fyne is designed to be platform independent.
But in my scenario I want to use it mainly on Windows and would like to make the window the topmost window. As it seems to me this is by default not possible (see https://github.com/fyne-io/fyne/issues/291) and I could achieve this by calling a certain Windows API function.
But this API needs the handle of the window and one approach to get this handle would be retrieving it from fyne

**Comment by andydotxyz** [comment]
Do you mean you want it to force to remain on top?

**Comment by mar1ged** [comment]
@andydotxyz Yes, exactly. I need to write create a go program that sits and waits in the background and opens a window to present information to the user. This window needs to be put on top of other open / running windows. We talked about this in the linked issue.

**Comment by andydotxyz** [comment]
Indeed, but from the usecase it sounds like a new window that gets focus will be sufficient - is that not how it works at the moment? If so then maybe there is a bug to be solved first.

**Comment by mar1ged** [comment]
@andydotxyz sorry, I don't get your point

**Comment by andydotxyz** [comment]
I guess I'm asking what about this does not currently work? New windows normally get focus.
So I'd like to know if you are not seeing that, and if we have a bug, before we add a new feature requirement.

**Comment by ohir** [comment]
@mar1ged From the thread(s) I've got an impression of XY problem.

> This window needs to be put on top of other open / running windows.

Unless there is an always on top window (other) any new window will appear focused on the top.

I suppose you're trying to implement a "blocker/curtain/nag" window, one that prevents user from interacting with other apps. - and **this is not possible in contemporary OSes** (iirc its even strictly forbidden in Apple's ecosystem). User always can use taskbar to override your hints, ie. detop then minimize your main window. 

PS. IIRC windows enterprise setup (in user profiles) was allowing for marked apps that can not be manipulated by user from the taskbar. Aka "curtains" or "gates" in (our) admins' lingo. With such setup your app might just make a maximized window, the rest is in the admin's hands.

**Comment by mar1ged** [comment]
@ohir Yes, this is to create a "nagscreen". In my case this is for a PC running on Windows in a kind of kiosk mode. The user is not able to return to the desktop or execute Windowskey+m. But he would be able to click on windows below " my" window, effectively bypassing the nagscreen or to put it more mildly accidentally missing the important info given by the screen.
There exists an Windows Api call that assigns the dialog the properties I need it to have. But for this API I need the handle.
So basically there are two approaches:
1. open the dialog with the correct put me on top attribute (see prior issue)
2. open the dialog, get the handle, modify the dialogs properties by using the api

**Comment by andydotxyz** [comment]
I’m guessing that adding a third option - make it full screen - is not useful?

This is a difficult one I admit, we don’t want to add APIs that are not functional on some systems (I.e. 1) but also not keen to expose internal pointers to platform specifics if it can be avoided.

It’s possible I suppose that there could be OS specific APIs that could handle things like this, like we have done with “driver/desktop” to provide hooks for desktop keyboard and mouse events?

**Comment by AndrewWPhillips** [comment]
I think this is handled in GLFW using the *GLFW_FLOATING* window creation hint - possibly works on all/most platforms.  Would it be possible to add "hints" to the NewWindow() method?

**Comment by andydotxyz** [comment]
Yes, but from their docs “ This is intended primarily for debugging purposes”... so I get the feeling they don’t really want to encourage it either. I’m not managing to see a clean way to do this, but am open to suggestions.

**Comment by andydotxyz** [comment]
Closing this issue - we're not going to expose the internal details unless it cannot be avoided.
For kiosk style software the operating system can make the window topmost and handle blocking user interaction with other parts, this is not the job of Fyne.

**Comment by matwachich** [comment]
For anyone who is wondering how to get HWND, [here is a working example](https://github.com/matwachich/fynex-widgets/blob/main/misc_windows.go#L26)
